core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -336,9 +336,9 @@ private[scheduler] object BlacklistTracker extends Logging {
    */
   def isBlacklistEnabled(conf: SparkConf): Boolean = {
     conf.get(config.BLACKLIST_ENABLED) match {
-      case Some(enabled) =>
+      case enabled =>
         enabled
-      case None =>
+      case _ =>
```

Comment:
Yeah, it would be only two values "true" or "false". But what I mean is that with this changes code will never touch this branch:

```
 // if they've got a non-zero setting for the legacy conf, always enable the blacklist,
        // otherwise, use the default.
        val legacyKey = config.BLACKLIST_LEGACY_TIMEOUT_CONF.key
        conf.get(config.BLACKLIST_LEGACY_TIMEOUT_CONF).exists { legacyTimeout =>
          if (legacyTimeout == 0) {
            logWarning(s"Turning off blacklisting due to legacy configuration: $legacyKey == 0")
            false
          } else {
            logWarning(s"Turning on blacklisting due to legacy configuration: $legacyKey > 0")
            true
          }
        }
```

I think we'd better keep the consistent semantics.